python/cudf/cudf/core/column/column.py

```diff
@@ -1219,7 +1219,7 @@ def _process_for_reduction(
             if result_col.has_nulls:
                 result_col = result_col.dropna()
         else:
-            if self.has_nulls:
+            if self.has_nulls or self.has_nans:
```

Comment:
Maybe we can move this bit of logic down to NumericalColumn and then call `super()._process_for_reduction`?

Reply:
Moved this to `NumericalColumn`